src/transforms/log_to_metric.rs

```diff
@@ -372,42 +415,417 @@ fn to_metric(config: &MetricConfig, event: &Event) -> Result<Metric, TransformEr
         .with_timestamp(timestamp))
 }
 
+fn bytes_to_str(value: &Value) -> Option<String> {
+    match value {
+        Value::Bytes(bytes) => std::str::from_utf8(bytes).ok().map(|s| s.to_string()),
+        _ => None,
+    }
+}
+
+fn try_get_string_from_log(log: &LogEvent, path: &str) -> Result<Option<String>, TransformError> {
+    // TODO: update returned errors after `TransformError` is refactored.
+    let maybe_value = log.parse_path_and_get_value(path).map_err(|e| match e {
+        PathParseError::InvalidPathSyntax { path } => FieldNotFound {
+            path: path.to_string(),
+        },
+    })?;
+    match maybe_value {
+        None => Err(FieldNotFound {
+            path: path.to_string(),
+        }),
+        Some(v) => Ok(bytes_to_str(v)),
+    }
+}
+
+fn get_counter_value(log: &LogEvent) -> Result<MetricValue, TransformError> {
+    let counter_value = log
+        .get(event_path!("counter", "value"))
```

Comment:
💭 TIOLI: 

Paths are hardcoded here and below. This is something that can be improved but needs a bit of refactoring:

- Add a `Option<Vec<ConfigTargetPath>>` to the config.
- The default values will be set if `all_metrics` is true and will be `["counter.value".into(), ...]` 
- Pass an new `impl TargetPath<'a>` argument to this function.

Reply:
Will make a feature request for this change.